src/layouts/sidebar-sidecar/utils/prepare-nav-data-for-client.ts

```diff
@@ -137,10 +137,18 @@ function prepareNavNodeForClient({
 			traversedNodes: traversedNodes + 1,
 		}
 	} else if (isNavLeaf(node)) {
-		// For nodes with paths, add fullPath to the node, and `id`
+		/**
+		 * For nodes with paths, add fullPath to the node, and `id`
+		 * Note: pathWithIndexFix is a stopgap, `index` items should
+		 * really be fixed up in content. At some point in the future,
+		 * we could add a warning or error here, or resolve this
+		 * through content conformance efforts.
+		 */
+		const pathWithIndexFix = node.path == 'index' ? '' : node.path
```

Comment:
Note: this felt worth adding, since the [/vagrant/docs](https://dev-portal-git-zsauto-overview-item-detect-hashicorp.vercel.app/vagrant/docs) `"Overview"` item doesn't highlight properly due to this issue. Added a note on this to [our content conformance doc](https://docs.google.com/document/d/1FQxRKiEDM5Kobm2MUSUnMC7ke-64yj5ceu6kDUpEiBI/edit).